src/sipmpdf/functions.py

```diff
@@ -218,31 +219,144 @@ def darkcurrent_response(x: np.float64,
 
 def ap_response_smeared(x: np.float64, smear: np.float64, n_ap: np.int64,
                         beta: np.float64) -> np.float64:
-  kern = kernel_switch(x, smear, n_np, beta)
-  pass
+  kern = kernel_switch(x, smear, n_ap, beta)
 
+  return kern.where(n_ap==1,
+                    kern.exp(-x/beta)*(1+kern.erf(x/(smear*kern.sqrt(2))))/(2*beta),
+                    ap_response(x=x,n_ap=n_ap,beta=beta))  
 
+@expand_shape
 def darkcurrent_response_smeared(x: np.float64,
                                  smear: np.float64,
-                                 pedestal: np.float64,
                                  gain: np.float64,
                                  resolution: np.float64 = 1e-4) -> np.float64:
-  kern = kernel_switch(x, smaer, pedestal, gain, resolution)
-  pass
+  kern = kernel_switch(x, smear, gain, resolution)
+
+  n_samples = 1024
 
+  #extend arrays
+  def extend_arr(x):
+    return kern.repeat(x[kern.newaxis,...], n_samples, axis=0)
 
+  x=extend_arr(x)
+  smear=extend_arr(smear)
+  gain=extend_arr(gain)
+  resolution=extend_arr(resolution)
+  
+  delta = 8 * smear / n_samples  # Getting the integration window
+  idx = kern.local_index(x, axis=0)
+  x_pr = -4 * smear + delta * (idx + 0.5)
+
+  # Given that the computed result will exhibit periodicity within the
+  # integration window, here we compute a simple rolling means with 5 parts
+  return kern.sum(darkcurrent_response(x - x_pr, gain, resolution)*normal(x_pr, 0, smear)*delta,axis=0)
+
+@expand_shape
+def afterpulsing_summation(x: np.float64,
+                           total: np.int64,
+                           ap_smear: np.float64, ##which should this actually be?
+                           ap_beta: np.float64, 
+                           ap_prob: np.float64,
+                           pedestal: np.float64,
+                           gain: np.float64) ->np.float64:
+  kern = kernel_switch(x, ap_smear, ap_prob, ap_beta, total,pedestal,gain)
+  k=10
+  ##come back to upgrade this later, decide on most effective method to ensure accuracy and not use too much computing power
+  def extend_arr(x):
+    return kern.repeat(x[kern.newaxis,...], k, axis=0)
+    
+  x=extend_arr(x)
+  ap_smear=extend_arr(ap_smear)
+  ap_beta=extend_arr(ap_beta)
+  ap_prob=extend_arr(ap_prob)
+  total=extend_arr(total)
+  pedestal=extend_arr(pedestal)
+  gain=extend_arr(gain)
+  
+  idx = kern.local_index(x, axis=0)
+  idx+=1
+
+  return kern.sum(binomial_prob(x=idx,total=total,prob=ap_prob)*ap_response_smeared(x=x-(pedestal+total*gain),smear=ap_smear,n_ap=idx,beta=ap_beta), axis=0)
+
+def k_summation(
+    x: np.float64,
+    common_noise: np.float64,
+    pixel_noise: np.float64,
+    ap_beta: np.float64,
+    ap_prob: np.float64,
+    pedestal: np.float64,
+    gain: np.float64,
+    poisson_mean: np.float64,
+    poisson_borel: np.float64)->np.float64:
+  kern = kernel_switch(x, common_noise, pixel_noise, ap_prob, ap_beta, pedestal, gain, poisson_mean, poisson_borel)
+  
+  k_max=10
```

Comment:
`k_max` should be set by `poisson_mean`. Look at the current `normal_smeared_poisson` for example.